### PR TITLE
Allow builtins in yul identifier paths in antlr grammar.

### PR DESCRIPTION
**Pull request by ekpyron**
Came up in https://github.com/ethereum/solidity/pull/12470

### CODE REVIEW

File: docs/grammar/SolidityParser.g4 [hunk 1]
```diff
@@ -564,7 +564,7 @@ yulFunctionDefinition:
  * While only identifiers without dots can be declared within inline assembly,
  * paths containing dots can refer to declarations outside the inline assembly block.
  */
-yulPath: YulIdentifier (YulPeriod YulIdentifier)*;
+yulPath: YulIdentifier (YulPeriod (YulIdentifier | YulEVMBuiltin))*;
```
Comment (ekpyron) [inline docs/grammar/SolidityParser.g4:567]: Strictly speaking probably ``address.test`` would in theory be a valid ``yulPath`` as well, but I'd say this hack is good enough for all intents and purposes.

**Comment by ekpyron** [comment]
If we want, we can think about whether we can do this "properly", i.e. have the solidity yul version just parse identifiers containing dots - but since this won't really happen too much in practice, since we reserved identifiers with dots for high-level-language specific constructions and we're unlikely to do more weird stuff like with ``functionPointer.address``, I'd say we should at least temporarily merge this, s.t. we can proceed with https://github.com/ethereum/solidity/pull/12470.

File: test/libsolidity/syntaxTests/inlineAssembly/assignment_to_function_pointer.sol [hunk 2]
```diff
@@ -0,0 +1,9 @@
+contract C {
+  function f() public pure {
+    function() external g;
+    assembly {
+      g.address := 0x42
+      g.selector := 0x23
+    }
+  }
+}
```
Comment (cameel) [inline test/libsolidity/syntaxTests/inlineAssembly/assignment_to_function_pointer.sol]:
```suggestion
contract C {
    function f() public pure {
        function() external g;
        assembly {
            g.address := 0x42
            g.selector := 0x23
        }
    }
}
```

**Review by cameel — DISMISSED** [review]
Just one cosmetic tweak needed so I'm approving already.

By the way. How does this affect recent efforts to specify Yul formally? Was that in any way dependent on our grammar definitions or done from scratch?

**Review by cameel — APPROVED** [review]

**Comment by ekpyron** [comment]
> By the way. How does this affect recent efforts to specify Yul formally? Was that in any way dependent on our grammar definitions or done from scratch?

I wouldn't be aware of the grammar being used for that (or actually for anything apart from our docs and tests for that matter).

**Comment by cameel** [comment]
Ok, good. Looking at the article that was posted last week ([Securing Warp: A formal specification of the Yul IR](https://medium.com/nethermind-eth/securing-warp-a-formal-specification-of-the-yul-ir-85bb3bf51c62)), they just [treat Yul identifiers as strings and don't have a "yul path" like we do in our grammar](https://github.com/NethermindEth/Yul-Specification/blob/main/yul_ast.lean).

**Comment by ekpyron** [comment]
Ah, that one you meant - that's a rather bold simplification of Yul anyways :-).

**Pull request merged**